core/templates/dev/head/components/state-editor/state-editor-properties-services/state-name.service.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A data service that stores the current state name.
+ */
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { AlertsService } from 'services/alerts.service';
+import { StatePropertyService } from
+  // eslint-disable-next-line max-len
+  'components/state-editor/state-editor-properties-services/state-property.service';
+import { UtilsService } from 'services/utils.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StateNameService extends StatePropertyService {
+  private stateNameEditorIsShown: boolean = false;
+  constructor(alertsService: AlertsService, utilsService: UtilsService) {
+    super(alertsService, utilsService);
+    this.setterMethodKey = 'saveName';
```

Comment:
Should this be stateName?

Also I am a little unsure about this change. The other StateProperties are stuff that are defined in the state dict, but this is the key corresponding to that dict. So, is it actually correct to treat this property just like the others?

Reply:
Hm, true. Thanks for pointing it out. I've refractored this a bit, PTAL!